cloudberry/zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/JSONParser.scala

```diff
@@ -56,6 +57,27 @@ object JSONParser {
     resolved
   }
 
+  def parseLimit(json: JsValue, option: QueryExeOption): (JsValue, QueryExeOption) = {
```

Comment:
this function contains too many nested if else. I would suggest rewriting the logic by adding `return`s to at least eliminate the nested branches.